setup.py

```diff
@@ -157,6 +157,14 @@ def configure(self, ext: CMakeExtension) -> None:
         # on subsequent calls to python.
         cmake_args += ['-DVLLM_PYTHON_PATH={}'.format(":".join(sys.path))]
 
+        # Override the base directory for FetchContent downloads to $ROOT/.deps
+        # This allows sharing dependencies between profiles,
+        # and plays more nicely with sccache.
+        # To override this, set the FETCHCONTENT_BASE_DIR environment variable.
+        fc_base_dir = os.path.join(ext.cmake_lists_dir, ".deps")
```

Comment:
Does `.deps` end up in the same place? It looks like `ext.cmake_lists_dir` is set to `.` by default but I'm not sure

Mainly asking to see if this change is going to cause any churn in developers' build processes, or in buildkite

Reply:
I assumed `ext.cmake_lists_dir` would refer to the root dir of the project (the directory where `CMakeLists.txt` resides).

Reply:
But if there's a better way for specifying the directory I'm happy to use that instead

Reply:
I don't see it getting set anywhere, but saw that it's an optional argument to `CMakeExtension`'s `__init__`

Reply:
Should I just use `ROOT_DIR` instead?

Reply:
yeah, let's go with that IMO